### DISCUSSION TITLE
Grid in plot

### DISCUSSION THREAD

**Discussion by dscso** [cetz-package/cetz#353] [Q&A] [answered]
Thank you very much for this awesome library.
I just have difficulties to get the grid in plots working. I tried multiple things:
```typst
#canvas(length: 1cm, {
  plot.plot(size: (8, 6),
    grid: true,
    x-tick-step: none,
    x-ticks: ((-calc.pi, $-pi$), (0, $0$), (calc.pi, $pi$)),
    y-tick-step: 1,
    {
      plot.add(
        domain: (-calc.pi, calc.pi), x => calc.sin(x * 1rad))
    })
})
```
<img width="276" alt="image" src="https://github.com/johannes-wolf/cetz/assets/6866008/c65e66f6-7f94-4ca7-9a9a-1cd4b0b072bf">

then I also tried this
```typst
#canvas(length: 1cm, {
  import draw: *
  set-style(axes: (
    grid: (
      stroke: (paint: luma(66.67%), dash: "dotted"),
      fill: red,)
    )
  )
  plot.plot(size: (8, 6),
    grid: true,
    x-tick-step: none,
    x-ticks: ((-calc.pi, $-pi$), (0, $0$), (calc.pi, $pi$)),
    y-tick-step: 1,
    {
      plot.add(
        domain: (-calc.pi, calc.pi), x => calc.sin(x * 1rad))
    })
})
```
I also tried the master branch, same result. Is there some simple way to activate the grid in the `plot` function?

**Comment by johannes-wolf** [comment, marked as answer]
Yes, you can activate the grid per axis via: `<axis>-grid: true`, e.G. `x-grid: true`. If you want to enable the grid for major and minor ticks, set `x-grid: "both"`.

![grafik](https://github.com/johannes-wolf/cetz/assets/519002/2c175247-45ba-4c77-9eba-d481cb82e97f)
